### ISSUE TITLE
Set visibility: hidden for tooltip before destroy if parent container of trigger was hidden

### ISSUE THREAD

**Issue by 7iomka** [bootstrap-vue/bootstrap-vue#4564]
### Is your feature request related to a problem? Please describe...

Currently, if we use the directive to initialize the tooltip, and do not specify a fixed container in the settings, the tooltip will be added to the end of the body and its position will be calculated as `top: 0, left: 0; transform: translate3d(calculatedValue);`

In the UI, you often have to use tooltips on buttons that hide (via display: none) their parent container.
After immediately hiding the trigger container on which the tooltip is initialized, we observe that the tooltip itself appears in the upper left corner of the screen before it is removed from the DOM. I examined the timeline in the chrome inspector, but I still do not understand the reason why this happens (`transform: translate3d(0, 0, 0)` positioning can be reset to zero but it happens so quickly that I don’t see it - I don’t know, tell me please if so).

### Describe the solution you'd like
Before destroy instance in described case, change css `visibility` to `hidden` for instance for preventing
display in the upper left corner of the screen.

### Describe alternatives you've considered
I'm currently using a rather nasty workaround:
1) I create an id for the trigger container (`containerId`) and specify it in the container option 
```
v-b-tooltip.hover="{container: 'containerId'}"
```
2) at the moment when the trigger parent must be hidden, for example, if there is a class .is-hidden
I write in css like this:
```
#containerId.is-hidden .trigger + .tooltip { display: none }
```
This is inconvenient because of the need to create identifiers or wrappers, as well as because of the complexity of handling some situations when one of the parents of the trigger is hidden. For example, when you hover the mouse over it, the styles will be such:
```
#containerId:not(:hover) .trigger + .tooltip { display: none; }
```

**Comment by tmorehouse** [comment]
The calculation of positioning is computed by Popper.js, and when the target is removed/hidden then those calculations for position become all zero.

Perhaps we could destroy the popper instance first to prevent the tooltip/popover from moving, but this can be a delicate dance, depending on timing of the `destroy` event fired by Vue.

**Comment by tmorehouse** [comment]
Or perhaps we need to add a visibility watcher on the container element (if not the default of `body`) and destroy popper and the tooltip.

We do have a watcher on the trigger element that periodically checks for the visibility of the trigger element, but it is not instantaneous.

**Comment by 7iomka** [comment]
It turns out for such a common case we currently have no suitable solution? :(

**Comment by ericdrobinson** [comment]
I ran into this issue something fierce. To work around it I manually set `display: none` on all Tooltips/Popovers that are within the hierarchy of objects removed (in my case they're under a `v-if` triggered by a mouse-out). To grab the Tooltips/Popovers, I supply them with `id` values and then query the document (or the container element, if one is set) to grab the root element upon which I can set the `style.display` property.

Without this, the reflow caused by elements moving to 0 positions happens every single time and I know of no other way to stop it (outside of possibly modifying BSV source).
